pkg/provisioner/ironic/ironic.go

```diff
@@ -1219,6 +1219,31 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	p.log.Info("requesting ports to delete")
+	port_list, err = ports.List(
+		p.client,
+		ports.ListOpts{NodeUUID: p.status.ID},
+	).Extract()
+	switch err.(type) {
+	case nil:
+		p.log.Info("port list obtained")
+	default:
+		return result, errors.Wrap(err, "failed to obtain port list")
+	}
+
+	for port_to_delete := range port_list {
```

Comment:
We could log which port we're removing before we try to remove it.

Reply:
Great idea!